operations/helm/charts/mimir-distributed/values.yaml

```diff
@@ -768,6 +774,8 @@ ingester:
     #   set, choosing the default provisioner.  (gp2 on AWS, standard on
     #   GKE, AWS & OpenStack)
     #
+    # This is ignored when ingester.zoneAwareReplication.enabled=true.
```

Comment:
What's the reason for having different behaviours when setting the storage class per zone vs for all zones?

Reply:
The reason is that I want to be able to have storage class per zone.  
If I don't, I only have 1/3 on ingester that start on each zone. 2/3 are pending waiting d for the PVC/PV to be ready. 
With this change I can force a storage class per zone and have all my ingester running

Reply:
Apologies, I didn't ask my question correctly. There are some special values for `storageClass`
* `null` leave the field empty, which lets k8s use the default provisioner in your cluster
* `"-"` leaves the field as `""`, which disables dynamic provisioning

I was suggesting to retain the behaviour for the per-zone storage class. Currently it's not possible to set `ingester.storageClass: "my-storage-class"` and `ingester.zoneAwareReplication.zones[1].storageClass: "-"`. The same outcome is still possible, but you'd have to disable dynamic provisioning at the top level and then override two of the zones to `"my-storage-class"`, which I think is less intuitive

Your implementation already uses `null` to fall back to the top level value. Do you think we can support the `"-"` special value in the per-zone configuration as well? We'll also have to document it in the comments if we choose to support it.

Reply:
But I totally agree that per-zone storage classes make sense for the chart

Reply:
Hi @dimitarvdimitrov and thanks for the comment, I updated the statefulsets if you can review it. Thanks.